types/ember-data/test/has-many.ts

```diff
@@ -59,3 +59,10 @@ declare module 'ember-data/types/registries/model' {
 class Polymorphic extends DS.Model {
     paymentMethods = DS.hasMany('payment-method', { polymorphic: true });
 }
+
+const comments = blogPost.hasMany('commentsAsync').value();
+if (comments) {
+    comments; // $ExpectType ManyArray<any>
+} else {
+    comments; // $ExpectType null
+}
```

Comment:
This can be written as
```ts
// $ExpectType ManyArray<any> | null
blogPost.hasMany('commentsAsync').value();
```
can you add a test case that goes through the model registry? i.e., a `ManyArray<User>` or something

Reply:
I've changed it but am not sure if there's a good way to test the relationship going through the model registry.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5f1f551c46d9ed1c3452fabd8eb504b7814779ca/types/ember-data/index.d.ts#L540 always returns a `HasManyReference<any>` when using `hasMany()` on a model.

I could try to create a `HasManyReference` manually but that wouldn't go through the model registry.